### PR TITLE
feat(api): get exams endpoint

### PR DESCRIPTION
**Pull request by Sembauke**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/how-to-open-a-pull-request/).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #XXXXX

### CODE REVIEW

File: api/src/exam-environment/routes/exam-environment.ts [hunk 1]
```diff
@@ -565,3 +572,28 @@ async function postScreenshotHandler(
...
+) {
+  const user = req.user!;
+  const exams = await this.prisma.envExam.findMany();
+
+  const availableExams = exams.map(exam => {
+    const isExamPrerequisitesMet = checkPrerequisites(user, true);
+
+    return {
+      examId: exam.id,
+      config: exam.config,
```
Comment (ShaunSHamilton) [inline api/src/exam-environment/routes/exam-environment.ts:589]: We do not want the whole config. Just the `name`, `note`, and `totalTimeInMS`.

File: api/src/exam-environment/routes/exam-environment.ts [hunk 2]
```diff
@@ -565,3 +572,28 @@ async function postScreenshotHandler(
...
+  reply: FastifyReply
+) {
+  const user = req.user!;
+  const exams = await this.prisma.envExam.findMany();
+
+  const availableExams = exams.map(exam => {
+    const isExamPrerequisitesMet = checkPrerequisites(user, true);
+
+    return {
+      examId: exam.id,
```
Comment (ShaunSHamilton) [inline api/src/exam-environment/routes/exam-environment.ts:588]: This is explicit enough, seeing as we are returning `exams[].id`

```suggestion
      id: exam.id,
```

File: api/src/exam-environment/schemas/exams.ts [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+  headers: Type.Object({
+    'exam-environment-authorization-token': Type.String()
+  }),
+  response: {
+    200: Type.Union([
+      Type.Object({
+        data: Type.Object({
+          exams: Type.Array(
+            Type.Object({
+              examId: Type.String(),
```
Comment (ShaunSHamilton) [inline api/src/exam-environment/schemas/exams.ts:13]:
```suggestion
              id: Type.String(),
```

File: api/src/exam-environment/routes/exam-environment.ts (hunk 1 above)
Comment (ShaunSHamilton) [reply api/src/exam-environment/routes/exam-environment.ts:589]: Use `select` in the `findMany` to ensure we are explicit about what is returned. Also, so the network request is smaller.

**Review by ShaunSHamilton — CHANGES_REQUESTED** [review]
Thanks for tackling this, Sem.

File: api/src/exam-environment/routes/exam-environment.test.ts [hunk 4]
```diff
@@ -524,6 +526,41 @@ describe('/exam-environment/', () => {
...
+
+    describe('GET /exam-environment/exams', () => {
+      it('should return 403', async () => {
+        const res = await superGet('/exam-environment/exams').set(
+          'exam-environment-authorization-token',
+          'invalid-token'
+        );
+
+        expect(res.status).toBe(403);
+      });
```
Comment (ShaunSHamilton) [inline api/src/exam-environment/routes/exam-environment.test.ts:538]: Last one: Please move this into the correct `describe` block below.

**Review by ShaunSHamilton — APPROVED** [review]
LGTM 👍

File: api/src/exam-environment/routes/exam-environment.test.ts [hunk 5]
```diff
@@ -524,15 +526,43 @@ describe('/exam-environment/', () => {
...
+
+    describe('GET /exam-environment/exams', () => {
+      it('should return 200', async () => {
+        const res = await superGet('/exam-environment/exams').set(
+          'exam-environment-authorization-token',
+          examEnvironmentAuthorizationToken
+        );
+        expect(res.status).toBe(200);
+
+        expect(res.body).toMatchObject({
```
Comment (ojeytonwilliams) [inline api/src/exam-environment/routes/exam-environment.test.ts:538]:
```suggestion
        expect(res.body).toStrictEqual({
```
Nit, strict is better, if possible.

File: api/src/exam-environment/routes/exam-environment.ts [hunk 6]
```diff
@@ -565,3 +572,37 @@ async function postScreenshotHandler(
...
+  const user = req.user!;
+  const exams = await this.prisma.envExam.findMany({
+    select: {
+      id: true,
+      config: true
+    }
+  });
+
+  const availableExams = exams.map(exam => {
+    const isExamPrerequisitesMet = checkPrerequisites(user, true);
```
Comment (ojeytonwilliams) [inline api/src/exam-environment/routes/exam-environment.ts:590]: I would inline it ` canTake: checkPrerequisites...`, but this is fine.

**Review by ojeytonwilliams — APPROVED** [review]
Just some nitpicks, so LGTM :+1:

**Pull request merged**